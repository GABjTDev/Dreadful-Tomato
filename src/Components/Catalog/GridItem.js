import React from 'react';
import CALENDAR from '../../assets/icon-calendar.png';


const GridItem = ({data}) => {
  return (
    <div className='grid--item'>
        <img src={data.images['Poster Art'].url} alt="dasdas" />
        <div className='grid--text'>
            <div className='grid--text-padding'>
                <h3>{data.title}</h3>
                <div className='grid--text-year'>
                    <img src={CALENDAR} alt='icon calendar' />
                    <p>{data.releaseYear}</p>
                </div>
                <p>
                    {data.description}
                </p>
            </div>
        </div>
    </div>
  )
}

export default GridItem