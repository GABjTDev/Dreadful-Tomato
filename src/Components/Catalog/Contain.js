import React, { useState, useEffect } from 'react';
import Loading from '../Loading/Loading';
import Pagination from '../Pagination/Pagination';
import GridItem from './GridItem';

const Contain = ({type, catalog}) => {

  const [loading, setLoading] = useState(true);

  useEffect(() => {

    setTimeout(() => {
      setLoading(false);
    }, 1000);

    return () => {
      clearTimeout();
      setLoading(true);
    };

  }, [catalog]);

  return (
    <div className='container'>
        <h2>Popular {type === 'movie' ? 'Movies': 'Series'}</h2>
        {
          loading ?
            <Loading />
          :
          (
            <>
              <div className='grid'>
                {
                  catalog.map(cat => <GridItem key={cat.title} data={cat}/>)
                }
              </div>
              <Pagination />
            </>
          )
        }
      
    </div>
  )
};

export default Contain;