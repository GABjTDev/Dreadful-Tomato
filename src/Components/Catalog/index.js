import React, {useEffect, useState} from 'react';
import { useParams } from 'react-router-dom';
import Header from '../Header';
import Search from '../Search/Search';
import data from '../../data/data.json';
import Contain from './Contain';

const Catalog = () => {

  const {type} = useParams();
  const [showSearch, setShowSearch] = useState(false);
  const [catalog, setCatalog] = useState([]);

  useEffect(() => {

    setShowSearch(false);
    // fetch('https://gitlab.com/-/snippets/2041384/raw/master/data.json')
    //   .then(res => res.json())
    //   .then(data => console.log(data))

    const dataFilter = data.entries.filter(el => el.programType === type).slice(0, 10);
    setCatalog(dataFilter);

  }, [type])
  

  return (
    <>
      <Header setShowSearch={setShowSearch} />
      <Search showSearch={showSearch} setCatalog={setCatalog} type={type} />
      <main className='main--catalog'>
        {
          catalog.length < 1 ?
            <p style={{color: 'white', textAlign: 'center'}}>There are no matches </p>
          :
          <Contain type={type} catalog={catalog} />
        }
      </main>
    </>
  )
}

export default Catalog