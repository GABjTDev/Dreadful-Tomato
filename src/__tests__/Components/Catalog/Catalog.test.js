import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter } from 'react-router-dom';
import Catalog from '../../../Components/Catalog';
import Contain from '../../../Components/Catalog/Contain';
import GridItem from '../../../Components/Catalog/GridItem';
import data from '../../../data/data.json';

describe('test catalog page', () => {

    test('Snapshot Catalog', () => {
        const history = createMemoryHistory();
        history.push('/catalog/movies');
        const wrapper = render(
            <BrowserRouter history={history}>
                <Catalog />
            </BrowserRouter>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Title Popular Movies', () => {
        const wrapper = render(
            <Contain type='movie' />
        );

        const titleMovies = wrapper.getByText(/popular movies/i);
        expect(titleMovies).toBeInTheDocument();
    });

    test('Title Popular Series', () => {
        const wrapper = render(
            <Contain type='series' />
        );

        const titleSeries = wrapper.getByText(/popular series/i);
        expect(titleSeries).toBeInTheDocument();
    });

    test('Draw gridItem with dats', () => {
        const wrapper = render(
            <GridItem data={data.entries[0]}/>
        );

        const titleSerie = wrapper.getByText('Wolf Creek').innerHTML;
        const yearMovie = parseInt(wrapper.getByText('2016').innerHTML);

        expect(titleSerie).toBe(data.entries[0].title);
        expect(yearMovie).toBe(data.entries[0].releaseYear);

    });

});