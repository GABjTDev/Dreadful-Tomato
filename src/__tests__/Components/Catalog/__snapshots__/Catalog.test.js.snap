// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test catalog page Snapshot Catalog 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="container header"
      >
        <div
          class="header--logo"
        >
          <a
            href="/"
          >
            <img
              alt="Logo de Dreadfull Tomato"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="header--login"
        >
          <ul>
            <li
              class="login"
            >
              Login
              <img
                alt="icon de login"
                src="icon-login.png"
              />
            </li>
            <li
              class="trialFree"
            >
              Star you free trial
            </li>
          </ul>
        </div>
        <div
          class="header--menu"
        >
          <img
            alt="icon de menu"
            src="icon-menu.png"
          />
        </div>
      </header>
      <div
        class="search false"
      >
        <div
          class="container search-container"
        >
          <form
            class="search--form"
          >
            <div
              class="input input--name"
            >
              <img
                alt="loupe icon"
                class="input--img"
                src="icon-loupe.png"
              />
              <input
                class="input--text"
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
            <div
              class="input input--picker"
            >
              <div
                class="input--click"
              >
                <img
                  alt="calendar icon"
                  class="input--img"
                  src="icon-calendar.png"
                />
                <input
                  class="input--text"
                  placeholder="1920-1988"
                  readonly=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              style="display: none;"
            />
          </form>
        </div>
      </div>
      <main
        class="main--catalog"
      >
        <p
          style="color: white; text-align: center;"
        >
          There are no matches 
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="container header"
    >
      <div
        class="header--logo"
      >
        <a
          href="/"
        >
          <img
            alt="Logo de Dreadfull Tomato"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="header--login"
      >
        <ul>
          <li
            class="login"
          >
            Login
            <img
              alt="icon de login"
              src="icon-login.png"
            />
          </li>
          <li
            class="trialFree"
          >
            Star you free trial
          </li>
        </ul>
      </div>
      <div
        class="header--menu"
      >
        <img
          alt="icon de menu"
          src="icon-menu.png"
        />
      </div>
    </header>
    <div
      class="search false"
    >
      <div
        class="container search-container"
      >
        <form
          class="search--form"
        >
          <div
            class="input input--name"
          >
            <img
              alt="loupe icon"
              class="input--img"
              src="icon-loupe.png"
            />
            <input
              class="input--text"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="input input--picker"
          >
            <div
              class="input--click"
            >
              <img
                alt="calendar icon"
                class="input--img"
                src="icon-calendar.png"
              />
              <input
                class="input--text"
                placeholder="1920-1988"
                readonly=""
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            style="display: none;"
          />
        </form>
      </div>
    </div>
    <main
      class="main--catalog"
    >
      <p
        style="color: white; text-align: center;"
      >
        There are no matches 
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
